src/Access/Access.php

```diff
@@ -249,7 +249,7 @@ public static function getAssetRules($asset, $recursive = false)
 		//$result = $query->rows()->fieldsByKey('rules');
 
 		// Get the root even if the asset is not found and in recursive mode
-		if (empty($result))
+		if (empty($result) && $recursive)
```

Comment:
This is the item I wasn't too sure about.  The Form\Fields\Rules class uses this method to generate existing permissions set on an item. For new items it was returning the root Asset even though recursive was not specified.  Therefore I changed this to only return the root asset when $recursive is set to true.